server/somesockets.py

```diff
@@ -1,47 +1,81 @@
-from flask import Flask, render_template
+from flask import Flask, render_template, request
 from flask_socketio import SocketIO, send, emit
+from secrets import SOCKETIO_SECRET_KEY
+from models import Session
+from flask_mongoengine import MongoEngine
 
+#=====================================================
+# Global vars
+#=====================================================
 app = Flask(__name__)
-app.config['SECRET_KEY'] = 'secret!'
+app.config['SECRET_KEY'] = SOCKETIO_SECRET_KEY
+app.config.from_object('config')
+
+db = MongoEngine(app)
+
 socketio = SocketIO(app)
+open_rooms = dict()
+bijection = dict()
 
 #=====================================================
 # SocketIO
 #=====================================================
-
-socketio = SocketIO(app)
-
 @socketio.on('connect')
 def handle_connect():
-    print("a user connected")
+    room = request.query_string.decode()
+    if room == "":
+        send('Malformed request')
+        return
+
+    session = Session.objects(code=room).first()
+    if session is None:
+        send('Room not found')
+        return
+
+    open_rooms[room] += [request.sid]
+    bijection[request.sid] = room
+    emit('Connect', dict(message="Another user connected"), room=room)
+
 
 @socketio.on('disconnect')
 def handle_disconnect():
-    print("a user has left")
+    room = bijection[request.sid]
+    open_rooms[room].remove(request.sid)
+    del bijection[request.sid]
+    emit('Disconnect', dict(message="User disconnected"), room=room)
+
 
 @socketio.on('update')
 def handle_update(json):
-    print('received json: ' + str(json))
-    emit('update', json)
+    try:
+        room = json['roomNumber']
+        map = json['map']
+    except:
+        send('Malformed request')
+        
+    emit('update', json=map, room=room)
 
-@socketio.on('FindRoom')
-def handle_FindRoom(json):
-    print('received room: ' + str(json))
     
-#TODO: Send this in the event the submitted room code isn't found
-def handle_RoomNotFound(json):
-    emit('RoomNotFound', json)
-    print ('Room Not Found!')
-
+@socketio.on('changeRoom')
```

Comment:
Can we fake namespace these? The event name could be something like "Session:ChangeMap"

Reply:
If we could store these strings in something like a constant global dictionary, that might be nice.

Reply:
i agree neil and I need to agree on a well-defined protocol